README.md

```diff
@@ -3,8 +3,257 @@ Nexus
+```
+$ cd nexus/apps/sample_app
+$ ./bin/simple_app -framework caffe2 -model vgg_face
+```
+These three programs construct a simple nexus, it can be tested by test_client.py in nexus/tests/python directory
+```
+$ cd nexus/tests/python
+$ python test_client.py
+```
+
+Build Docker Image
+------------------
+
+Since we recommand to use docker, there are four docker images that needs to be built.
+
+First you need to build the base docker image that installs all dependent libraries required for Nexus.
+
+Following docker build commands contain -t and -f options. -t option is followed 
+by name and optionally a tag in the ‘name:tag’ format. -f option is followed by 
+name of the Dockerfile (Default is ‘PATH/Dockerfile’).
+
+```
+$ cd nexus/dockerfiles
+$ docker build -t nexus/base -f NexusBaseDockerfile .
+```
+
+Next we can build the docker images for backend, scheduler, and application
+library respectively.
+```
+$ docker build -t nexus/backend -f NexusBackendDockerfile .
+$ docker build -t nexus/scheduler -f NexusSchedulerDockerfile .
+$ docker build -t nexus/applib -f NexusAppLibDockerfile .
+```
+
+### Compile sample applications
+
+In the Nexus repo, we provide a few sample applications located at `nexus/apps`.
+In each application directory, you can build the application docker image by
+```
+$ docker build -t nexus/app_name -f Dockerfile .
+```
+
+Nexus Deployment
+----------------
+
+### Step 1: Create a virtual cluster (optional)
+If you only need to deploy Nexus on a single server, you can skip this step.
+To run Nexus service across multiple servers, we use the docker swarm (requires
+docker version >= 1.20) to manage a cluster.
+
+#### Add servers to swarm cluster
+
+First create the swarm on a master server, run
+```
+$ docker swarm init
+```
+This server serves as a manager node of the swarm.
+
+To add worker nodes to this swarm, we need to get the token by running command
+on the master node
+```
+$ docker swarm join-token worker
+```
+The example output of this command is
+```
+$ docker swarm join \
+--token SWMTKN-1-49nj1cmql0jkz5s954yi3oex3nedyz0fb0xx14ie39trti4wxv-8vxv8rssmk743ojnwacrr2e7c \
+192.168.99.100:2377
+```
+Copy this piece of code and run it on the worker server.
+
+To check nodes and their roles in the swarm, run the following command on a manager node
+```
+$ docker node ls
+```
+
+#### Create overlay network
+
+After the swarm is created, we then need to create a overlay network to allow
+dockers in the swarm to communicate with each other. Run the following command on
+a manager node.
+```
+$ docker network create --driver overlay --attachable --subnet 10.0.0.0/16 nexus-network
+```
+
+### Step 3: Generate model profile
+
+In order to evaluate the maximum batch size for each model in each GPU, we need to generate profiles by profiler in tools directory. 
+```
+$ cd nexus/tools/profiler
+$ docker run --runtime=nvidia -t -v /path/to/nexus-models:/nexus-models -v /path/to/dataset:/dataset:ro nexus/backend \
```

Comment:
Let's put two approaches to use profiler:

If you build nexus on your local machine, you can launch profiler by
```
$ cd /path/to/nexus/tools/profiler
$ python profiler.py $(framework) $(model) /path/to/nexus-models /path/to/dataset
```
Otherwise, you can launch profiler through docker
```
docker run ......
```